sentry_sdk/integrations/gql.py

```diff
@@ -0,0 +1,170 @@
+    try:
+        return transport.method
+    except AttributeError:
+        return "POST"
+
+
+def _request_info_from_transport(transport):
+    # type: (Union[Transport, AsyncTransport, None]) -> Dict[str, str]
+    if transport is None:
+        return {}
+
+    request_info = {
+        "method": _transport_method(transport),
+    }
+
+    try:
+        request_info["url"] = transport.url
+    except AttributeError:
+        pass
+
+    return request_info
+
+
+def _patch_execute():
+    # type: () -> None
+    real_execute = gql.Client.execute
+
+    def sentry_patched_execute(self, document, *args, **kwargs):
+        # type: (gql.Client, DocumentNode, Any, Any) -> Any
+        hub = Hub.current
+        if hub.get_integration(GQLIntegration) is None:
+            return real_execute(self, document, *args, **kwargs)
+
+        with _graphql_event_processing(self, document):
+            try:
+                return real_execute(self, document, *args, **kwargs)
+            except TransportQueryError as e:
+                event, hint = event_from_exception(
+                    e,
+                    client_options=hub.client.options
+                    if hub.client is not None
+                    else None,
+                    mechanism={"type": "gql", "handled": False},
+                )
+
+                hub.capture_event(event, hint)
+                raise e
+
+    gql.Client.execute = sentry_patched_execute
+
+
+def _make_gql_event_processor(client, document):
+    # type: (gql.Client, DocumentNode) -> EventProcessor
+    def processor(event, hint):
+        # type: (Dict[str, Any], Dict[str, Any]) -> Dict[str, Any]
+        try:
+            errors = hint["exc_info"][1].errors
+        except (AttributeError, KeyError):
+            errors = None
+        _deep_update(
+            event,
+            {
+                "request": {
+                    "api_target": "graphql",
+                    **_request_info_from_transport(client.transport),
+                },
+            },
+        )
+
+        if _should_send_default_pii():
+            _deep_update(
+                event,
+                {
+                    "request": {
+                        "data": _data_from_document(document),
+                    },
+                    "contexts": {
+                        "response": {
+                            "data": {
+                                "errors": errors,
+                            },
+                            "type": "response",
+                        },
+                    },
+                },
+            )
+
+        return event
+
+    return processor
+
+
+@contextmanager
+def _graphql_event_processing(client, document):
+    # type: (gql.Client, DocumentNode) -> Generator[None, None, None]
+
+    with Hub.current.configure_scope() as scope:
+        scope.add_event_processor(_make_gql_event_processor(client, document))
+        yield
+
+
+def _deep_update(dictionary, update):
```

Comment:
This feels like a more general purpose thing, would it make sense to move this to `utils.py`? It also feels like we should have some unit tests just for this.